internal/services/containerapps/container_app_environment_resource_test.go

```diff
@@ -455,6 +481,27 @@ resource "azurerm_subnet" "control" {
 `, r.template(data), data.RandomInteger)
 }
 
+func (r ContainerAppEnvironmentResource) templateVnetSubnetNotDelegated(data acceptance.TestData) string {
```

Comment:
Subnets are supposed to need delegation for Environments to use them according to the docs afaik, what's the reasoning for this needing it removed?

Reply:
A delegated subnet cannot be used for creating container app environment according to the API response. Here's the error message if it's delegated:
```
Status: "Failed"
Code: "ManagedEnvironmentSubnetIsDelegated"
Message: "AgentPoolProfile subnet with id xxxx cannot be used as it's a delegated subnet. Please check https://aka.ms/adv-network-prerequest for more details.\r\nStatus: 400 (Bad Request)\r\nErrorCode: SubnetIsDelegated\r\n\r\nContent:\r\n{\n  \"code\": \"SubnetIsDelegated\",\n  
```